src/libsystemd/sd-journal/journal-authenticate.c

```diff
@@ -76,7 +78,11 @@ int journal_file_hmac_start(JournalFile *f) {
         /* Prepare HMAC for next cycle */
         gcry_md_reset(f->hmac);
         FSPRG_GetKey(f->fsprg_state, key, sizeof(key), 0);
-        gcry_md_setkey(f->hmac, key, sizeof(key));
+        err = gcry_md_setkey(f->hmac, key, sizeof(key));
+        if (gcry_err_code(err)) {
+                log_debug("gcry_md_setkey() failed with error code: %d", gcry_err_code(err));
+                return -1;
```

Comment:
No fake return codes please, use a real errno. And log_debug_errno(SYNTHETIC_ERRNO()).

Reply:
The problem is that there's no errno :-/ this library has its own return codes, which overlap with errnos but do not match (as far as I understand):

https://gnupg.org/documentation/manuals/gcrypt/Error-Codes.html#Error-Codes

Reply:
and as far as I understand it means I can't just return it, right? Otherwise it will be mis-interpreted.

Reply:
Yes, that's why SYNTHETIC_ERRNO. Just pick something that make sense, ENOMEM, or EIO, etc.

Reply:
Ah sorry, misunderstood - done